demo-project/generic/build.gradle.kts

```diff
@@ -56,8 +56,16 @@ val generateBuildResourcesBuildConfigTest = task<AssertGeneratedFile>("generateB
         """)
 }
 
-tasks.register("test") {
-    dependsOn(generateBuildConfigTest, generateBuildResourcesBuildConfigTest)
+tasks {
+
+    val test by registering {
+        dependsOn(generateBuildConfigTest, generateBuildResourcesBuildConfigTest)
+    }
+
+    build {
+        dependsOn(test)
+    }
```

Comment:
The missing `build` dependency on `test` caused theses "tests" not to run on CI and were broken